vbfml/scripts/preprocess_image.py

```diff
@@ -0,0 +1,431 @@
+    batch_size = 5000
+    batch_number = ceil(tree.num_entries / batch_size)
+    batch_counter = 0
+
+    # iterate by batches of 500 over the tree, aim to save memoryquit
+    for sub_tree in tree.iterate(
+        ["JetImage_pixels", "leadak4_phi", "leadak4_eta"],
+        step_size=batch_size,
+        library="np",
+    ):
+
+        batch_counter += 1
+        print(f"batch {batch_counter}/{batch_number}")
+
+        batch_index = (batch_counter - 1) * batch_size
+
+        # initialize the new array for the preprocess images
+        New_images = np.zeros((batch_size, n_eta_bins * n_phi_bins), dtype="uint8")
+
+        # fix size of last batch
+        if batch_counter == batch_number:
+            batch_size = tree.num_entries % batch_size
+            New_images = np.zeros((batch_size, n_eta_bins * n_phi_bins), dtype="uint8")
+
+        New_images_batch = np.ones((batch_size, n_eta_bins, n_phi_bins))
+        for j in range(batch_size):
+            New_images_batch[j] = sub_tree["JetImage_pixels"][j].reshape(im_shape)
+
+            # set phi = 0 (leading jet is at the center horizontal)
+            for i in range(n_eta_bins):
+                shift_phi = -round(
+                    sub_tree["leadak4_phi"][j] * n_phi_bins / (2 * np.pi)
+                )
+                New_images_batch[j][i] = np.roll(New_images_batch[j][i], shift_phi)
+
+            # set eta > 0 (leading jet is on the right side)
+            if sub_tree["leadak4_eta"][j] < 0:
+                New_images_batch[j] = np.flip(New_images_batch[j], 0)
+
+            New_images[j] = New_images_batch[j].flatten()
+
+        # writing in the new tree
+        new_tree = {}
+        new_tree["JetImage_pixels_preprocessed"] = New_images  # add the new branch
+        for branch in tree.keys():  # copy all the other branches
+            new_tree[branch] = tree[branch].arrays(
+                entry_start=batch_index,
+                entry_stop=batch_index + batch_size,
+                library="np",
+            )[branch]
+
+        if batch_counter == 1:
+            file["sr_vbf"] = new_tree
+        else:
+            file["sr_vbf"].extend(new_tree)
+
+        print(file["sr_vbf"].num_entries)
+
+
+@cli.command()
+@click.option(
+    "-i",
+    "--input-dir",
+    required=True,
+    help="Path to the directory with the input ROOT files.",
+)
+def rotate_all(input_dir: str):
+    """
+    Preprocess all root files of the directory by applying rotate function on every files of the input_dir
+    Warning -> have weird behavior over a lot of files. miss some events
+    """
+    files = glob.glob(pjoin(input_dir, f"*root"))
+
+    for file in tqdm(files, desc="Rotating images"):
+        print(
+            f"/////////////////// \nprocessing {os.path.basename(file)}\n///////////////////"
+        )
+        os.system("./preprocess_image.py rotate -i " + file)
+
+
+@cli.command()
+@click.option(
+    "-i",
+    "--input-files",
```

Comment:
Similar to the `rotate` function, if this function only accepts a single ROOT file, can you rename this to `input-file` as well to make it clear? And if I understand correctly, the `help` message here could be something like "Path to the input ROOT file" (as opposed to a directory)